### PR TITLE
Remove HipChat Notification Type

### PR DESCRIPTION
**Pull request by beeankha**
##### SUMMARY

[HipChat notifications have reached EoL](https://www.atlassian.com/partnerships/slack/faq#faq-98b17ca3-247f-423b-9a78-70a91681eff0):

image

This PR will remove HipChat notification types from the available `notification_type` choices.

In the `tower_notifications` Collections module, the `api_url` and `rooms` params will emit a different warning that specifically informs the user that HipChat is no longer a valid notification. 

##### AWX VERSION

```
awx 11.2.0
```

### CODE REVIEW

**Review by ryanpetrello — APPROVED** [review]
This looks awesome - thanks for doing this @beeankha!

**Comment by ryanpetrello** [comment]
cc @tvo318 if we have any official docs that reference HipChat, it's probably time to officially remove them.

**Comment by ryanpetrello** [comment]
@beeankha would you start a new block at https://github.com/ansible/awx/blame/devel/CHANGELOG.md#L5 e.g.,

```
12.0.0 (TBD)
- Removed support for HipChat notifications (including URL to their EOL announcement)
```

Thanks!

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 This is a list of high-level changes for each release of AWX. A full list of commits can be found at `https://github.com/ansible/awx/releases/tag/<version>`.
 
+## 12.0.0 (TBD)
+- Removed support for HipChat notifications (see the related EoL announcement [here](https://www.atlassian.com/partnerships/slack/faq#faq-98b17ca3-247f-423b-9a78-70a91681eff0))
```
Comment (ryanpetrello) [inline CHANGELOG.md]: a little nitpicky 😄 , but I don't like clickable  "here" links, maybe:

`[EoL announcement](https://www.atlassian.com/partnerships/slack/faq#faq-98b17ca3-247f-423b-9a78-70a91681eff0)`

**Comment by ryanpetrello** [comment]
@AlanCoding and @wenottingham can you weigh on this statement?  I'm having trouble deciding how I feel about it.

If you have a `NotificationTemplate` that is `notification_type=hipchat`, we *must* delete it, because it no longer makes sense.

If the template _itself_ goes away, we should probably *also* cascade delete the notification records themselves, right (at least, this seems to be how it has *always* worked)?  Do you see any problems with that?

**Comment by wenottingham** [comment]
If we previously cascade delete notifications when their templates go away, we should do that.
If we *don't* do that now, we shouldn't do that as part of this change.

**Comment by beeankha** [comment]
> If we previously cascade delete notifications when their templates go away, we should do that.

Yep, previous (current `devel`) setup is a cascade delete of notifications.  Will revert my changes so it goes back to that structure!

**Review by ryanpetrello — APPROVED** [review]
Okay, my 👍 on this still stands.  @AlanCoding, you have any opinions?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 This is a list of high-level changes for each release of AWX. A full list of commits can be found at `https://github.com/ansible/awx/releases/tag/<version>`.
 
+## 12.0.0 (TBD)
+- Removed support for HipChat notifications ([EoL announcement](https://www.atlassian.com/partnerships/slack/faq#faq-98b17ca3-247f-423b-9a78-70a91681eff0))
```
Comment (AlanCoding) [inline CHANGELOG.md]: might want to mention that any hipchat stuff users have will be deleted. Just get that out in the open.

File: awx_collection/plugins/modules/tower_notification.py [hunk 3]
```diff
@@ -159,7 +158,7 @@
     api_url:
       description:
         - The HipChat API URL.
-        - This parameter has been deprecated, please use 'notification_configuration' instead.
+        - The HipChat notification type has been deprecated.
```
Comment (AlanCoding) [inline awx_collection/plugins/modules/tower_notification.py]: I don't see why we need to keep this at all, if it's hipchat-only isn't the best solution to just delete it?

Comment (beeankha) [reply awx_collection/plugins/modules/tower_notification.py]: I was thinking of this as kind of a "soft fail" but maybe it's muddying the waters a bit.  I can go ahead and remove these parameters.

Comment (AlanCoding) [reply awx_collection/plugins/modules/tower_notification.py]: I can see the merit in leaving a tombstone, but in that case I still think `module.fail_json` would be better than `module.deprecate` that it does now.

Comment (beeankha) [reply awx_collection/plugins/modules/tower_notification.py]: There's actually no way to get to these `module.` warnings outside of a testing scenario, so I'm going to just go ahead and remove everything related to HipChat in the `tower_notification` module.

**Review by AlanCoding — APPROVED** [review]

**Pull request merged**